### ISSUE TITLE
@StorageProperty not resubscribing when coming from a route

### ISSUE THREAD

**Issue by delasteve** [SirDarquan/h5webstorage#5]
I noticed a bug when switching routes. I am using the [angular-cli](https://cli.angular.io/) to scaffold my project.

Reproduce steps:
1. Clone https://github.com/delasteve/h5webstorage-bug
2. `npm install`
3. `ng serve`
4. Click on `Store 1` and `Store 2` (notice it changes)
5. Click `Other Home`
6. Click `Home`
7. Select the not stored value button (notice no change)

As an added bonus, if you go back to `Other Home` and back to `Home`, the value updates properly, but again, the functionality of updating the value does not.

**Comment by SirDarquan** [comment]
Thanks for the feedback @delasteve. So what I noticed first was the way you were using the @StorageProperty. The idea was that the property you added the decorator was IS the value in storage so you won't have to initialize it in the constructor or in the $onInit method. But once I "corrected" your code to the way I'd intended it to be used, I noticed an initialization error. The initializer was overwriting the stored value which isn't supposed to happen. Here's the code as it should be and I'll work on the initialization bug to actually make it work properly.

``` typescript
import { Component, OnInit } from '@angular/core';
//Use the LOCAL_STORAGE_PROVIDER like ROUTE_PROVIDER, in the root component
import { LocalStorage, StorageProperty } from 'h5webstorage';

@Component({
  moduleId: module.id,
  selector: 'app-home',
  templateUrl: 'home.component.html',
  styleUrls: ['home.component.css']
})
export class HomeComponent implements OnInit {

  @StorageProperty() public valInStorage: number = 1; // this is supposed to init to 1 ONLY IF there is no value in storage

  constructor(private localStorage: LocalStorage) {}

  ngOnInit() {
    //this.valInStorage = this.localStorage['valInStorage']; //this is not necessary
  }

  clicked(valToStore) {
    this.valInStorage = valToStore; //assigning to your property assigns directly to storage
  }
}
```

I've been meaning to put up a really good example explaning this. I guess I need to so it sooner rather than later

**Comment by delasteve** [comment]
`@StorageProperty() public valInStorage: number = 1;` is what I expected to have happen as you mention it in the example app: https://github.com/SirDarquan/h5webstorage/blob/master/example/app/index.ts#L24

Thanks for the tips on how to clean up my code. Also, for the `LOCAL_STORAGE_PROVIDER`, I was just following your example...
https://github.com/SirDarquan/h5webstorage/blob/master/example/index.ts#L5
https://github.com/SirDarquan/h5webstorage/blob/master/example/app/index.ts#L21

As for the other bug, glad I could help you find it 😄. I'll be patiently awaiting the next release.

Thanks for this!

**Comment by SirDarquan** [comment]
Ok, just to be a little transparent, I've found the cause of the issue but I don't know if this behavior is going to change in the future. It seems when an object is created , it's placed in the injection heirarchy for later use. The very first time it is created, I can tell because a variable I create named 'initializing' is false (as is all boolean variable when first declared). It goes through some initialization stuff and then it's ready. Leaving a view does not necessarily destroy the object and if the view is revisited the already instantiated object is reused but for some strange reason the initialization value is resent. This time my 'initializing' variable is false and that allows the value to be set overwriting the stored value. This may take a awhile to work around and to come up with a test that ensures the behavior is intact.

As for the code suggestions, I'm still figuring out best practices with this module also and the advice I gave here is the newest ways I've come up with. I just haven't places that in the examples yet.
